### ISSUE TITLE
Optional node-input in rewriter pattern

### ISSUE THREAD

**Issue by gramalingam** [microsoft/onnxscript#1385]
What does it mean when one of the node-inputs is "" (when the pattern is described using a FunctionProto)?

It seems best to treat it as a requirement that the corresponding input in a graph must also be "" for a successful match.

Another scenario is where the user does not care what that corresponding node-input is (for a match). This must be represented by using a new function-parameter.

**Comment by justinchuby** [comment]
I agree that having a special wildcard construct to mean “don’t care” is preferable
